subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CopyPermissionsIntegrationTest.groovy

```diff
@@ -280,27 +280,27 @@ class CopyPermissionsIntegrationTest extends AbstractIntegrationSpec {
             task copy(type: Copy) {
                 from '${input.name}'
                 into '${outputDirectory.name}'
+                doNotTrackDestinationDir()
```

Comment:
I think we should keep the existing test with the deprecation warning, and add a new test with `doNotTrackDestinationDir()` instead? Especially that we talked about adding a more specific warning for `Copy`. We should also consider a more specific warning for `Sync`.

Reply:
I changed the tests so they check for the deprecation warning (a specific one for Copy) and call the method and copy and see if the deprecation warning goes away.

I also added a specific warning for Sync and a test for it.